File: backend-java/src/main/java/com/bffgin/backend/domain/TaskStatus.java
package com.bffgin.backend.domain;

import java.util.Optional;

/** backend(Go)のTaskStatus(waiting=1, work_in_progress=2, completed=3)と同じマッピング */
public enum TaskStatus {
    WAITING(1, "waiting"),
    WORK_IN_PROGRESS(2, "work_in_progress"),
    COMPLETED(3, "completed");

    private final int dbValue;
    private final String wireValue;

    TaskStatus(int dbValue, String wireValue) {
        this.dbValue = dbValue;
        this.wireValue = wireValue;
    }

    public int dbValue() {
        return dbValue;
    }

    public String wireValue() {
        return wireValue;
    }

    public static Optional<TaskStatus> fromWireValue(String s) {
        for (TaskStatus status : values()) {
            if (status.wireValue.equals(s)) {
                return Optional.of(status);
            }
        }
        return Optional.empty();
    }

    public static Optional<TaskStatus> fromDbValue(int v) {
        for (TaskStatus status : values()) {
            if (status.dbValue == v) {
                return Optional.of(status);
            }
        }
        return Optional.empty();
    }
}

File: backend-kotlin/src/main/kotlin/com/bffgin/backend/domain/TaskStatus.kt
package com.bffgin.backend.domain

/** backend(Go)のTaskStatus(waiting=1, work_in_progress=2, completed=3)と同じマッピング */
enum class TaskStatus(val dbValue: Int, val wireValue: String) {
    WAITING(1, "waiting"),
    WORK_IN_PROGRESS(2, "work_in_progress"),
    COMPLETED(3, "completed");

    companion object {
        fun fromWireValue(s: String): TaskStatus? = entries.firstOrNull { it.wireValue == s }
        fun fromDbValue(v: Int): TaskStatus? = entries.firstOrNull { it.dbValue == v }
    }
}
